shared/src/main/java/org/odk/collect/shared/PathUtils.kt

```diff
@@ -10,4 +13,16 @@ object PathUtils {
     // https://stackoverflow.com/questions/2679699/what-characters-allowed-in-file-names-on-android
     @JvmStatic
     fun getPathSafeFileName(fileName: String) = fileName.replace("[\"*/:<>?\\\\|]".toRegex(), "_")
```

Comment:
I think we should lock this behaviour in with a test if we're confident about it now.

Reply:
What do you mean? This method has nothing to do with the issue we had with paths.

Reply:
Sorry! I meant to comment on `getAbsoluteFilePath`.

Reply:
This wouldn't be easy. We have tests for `#sanitizedCanonicalPath` we can't write the same tests for `#getAbsoluteFilePath` because the values we pass to `#getAbsoluteFilePath` are not broken. The problem was with `#canonicalPath` called by `#getAbsoluteFilePath` (then broken paths ocurred), so I replaced it with `#sanitizedCanonicalPath` and that one is covered with tests.